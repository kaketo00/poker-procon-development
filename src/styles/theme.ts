import { extendTheme } from '@chakra-ui/react';
import { Zen_Kaku_Gothic_New } from 'next/font/google';

const nextFont = Zen_Kaku_Gothic_New({
  weight: ['400', '700', '900'],
  subsets: ['latin'],
});

const theme = extendTheme({
  fonts: {
    heading: nextFont.style.fontFamily,
    body: nextFont.style.fontFamily,
  },
  colors: {
    brand100: {
      100: '#0DD203', // green 100%
      200: '#005CFF', // blue 100%
      300: '#B4171E', // red 100%
      600: '#777777', // dark-gray 100%
      700: '#D4D4D4', // gray 100%
      800: '#ECECEC', // light-gray 100%
    },
    brand80: {
      100: 'rgba(13, 210, 3, 0.8)', // green 80%
      200: 'rgba(0, 92, 255, 0.8)', // blue 80%
      300: 'rgba(180, 23, 30, 0.8)', // red 80%
      400: 'rgba(255, 255, 255, 0.8)', // white 80%
      500: 'rgba(0, 0, 0, 0.8)', // black 80%
      600: 'rgba(119, 119, 119, 0.8)', // gray 80%
    },
  },
  textStyles: {
    normal: {
      xs: {
        fontSize: '0.625rem',
        fontWeight: 'normal',
        lineHeight: '0.75rem',
      },
      sm: {
        fontSize: '0.75rem',
        fontWeight: 'normal',
        lineHeight: '1rem',
      },
      md: {
        fontSize: '1rem',
        fontWeight: 'normal',
        lineHeight: '1.25rem',
      },
      lg: {
        fontSize: '1.25rem',
        fontWeight: 'normal',
        lineHeight: '1.5rem',
      },
      xl: {
        fontSize: '1.5rem',
        fontWeight: 'normal',
        lineHeight: '1.75rem',
      },
    },
    bold: {
      xs: {
        fontSize: '0.625rem',
        fontWeight: 'bold',
        lineHeight: '0.75rem',
      },
      sm: {
        fontSize: '0.75rem',
        fontWeight: 'bold',
        lineHeight: '1rem',
      },
      md: {
        fontSize: '1rem',
        fontWeight: 'bold',
        lineHeight: '1.25rem',
      },
      lg: {
        fontSize: '1.25rem',
        fontWeight: 'bold',
        lineHeight: '1.5rem',
      },
      xl: {
        fontSize: '1.5rem',
        fontWeight: 'bold',
        lineHeight: '1.75rem',
      },
    },
    strong: {
      xs: {
        fontSize: '0.6875rem',
        fontWeight: 'black',
        lineHeight: '1rem',
      },
      sm: {
        fontSize: '1.25rem',
        fontWeight: 'black',
        lineHeight: '1.5rem',
      },
      md: {
        fontSize: '1.5rem',
        fontWeight: 'black',
        lineHeight: '2rem',
      },
      lg: {
        fontSize: '1.875rem',
        fontWeight: 'black',
        lineHeight: '2.375rem',
      },
      xl: {
        fontSize: '2.25rem',
        fontWeight: 'black',
        lineHeight: '2.75rem',
      },
    },
    heading: {
      h6: {
        fontSize: '1rem',
        fontWeight: 'black',
        lineHeight: '1.25rem',
      },
      h5: {
        fontSize: '1.25rem',
        fontWeight: 'black',
        lineHeight: '1.5rem',
      },
      h4: {
        fontSize: '1.5rem',
        fontWeight: 'black',
        lineHeight: '1.875rem',
      },
      h3: {
        fontSize: '1.75rem',
        fontWeight: 'black',
        lineHeight: '2.125rem',
      },
      h2: {
        fontSize: '2.25rem',
        fontWeight: 'black',
        lineHeight: '2.75rem',
      },
      h1: {
        fontSize: '1.875rem',
        fontWeight: 'black',
        lineHeight: '2.25rem',
      },
    },
    shadows: {
      light5: {
        100: '0px 0px 5px rgba(13, 210, 3, 0.8)',
        200: '0px 0px 5px rgba(0, 92, 255, 0.8)',
        300: '0px 0px 5px rgba(180, 23, 30, 0.8)',
        400: '0px 0px 5px rgba(255, 255, 255, 0.8)',
      },
      light10: {
        100: '0px 0px 10px rgba(13, 210, 3, 0.8)',
        200: '0px 0px 10px rgba(0, 92, 255, 0.8)',
        300: '0px 0px 10px rgba(180, 23, 30, 0.8)',
        400: '0px 0px 10px rgba(255, 255, 255, 0.8)',
      },
    },
  },
  breakpoints: {
    base: '0px', // 0 - 1279px game-template(S)
    sm: '80em', // 1280px - 1439px game-template(S)
    md: '90em', // 1440px - 1919px game-template(M)
    lg: '120em', // 1920px - 2559px game-template(M)
    xl: '160em', // 2560px - âˆž game-template(L)
    '2xl': '160em',
  },
});

export default theme;
