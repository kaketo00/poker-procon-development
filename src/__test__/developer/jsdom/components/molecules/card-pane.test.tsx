import { ChakraProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';

import type { CardPanelProps } from '@/components/molecules/card-panel';
import CardPanel from '@/components/molecules/card-panel';
import ROUTE from '@/constants/route';
import theme from '@/styles/theme';

const Component = (props: CardPanelProps) => (
  <ChakraProvider theme={theme}>
    <CardPanel {...props} />
  </ChakraProvider>
);

describe('CardPanel', () => {
  test('スナップショット/default', () => {
    const { asFragment } = render(<Component number={1} index={1} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/md', () => {
    const { asFragment } = render(<Component number={1} index={1} size="sm" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/md', () => {
    const { asFragment } = render(<Component number={1} index={1} size="md" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/link', () => {
    const { asFragment } = render(
      <Component number={1} index={1} href={ROUTE.S_A01_001} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
