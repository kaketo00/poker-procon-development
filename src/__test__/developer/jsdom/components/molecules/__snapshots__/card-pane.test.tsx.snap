// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardPanel スナップショット/default 1`] = `
<DocumentFragment>
  <div
    class="css-vxlbt5"
  >
    <div
      class="css-u8yv6t"
    >
      <p
        class="chakra-text css-bq8qtg"
      >
        1
      </p>
      <div
        class="css-14d7ov9"
      >
        <svg
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 35.005L34.2046 20.2998C37.9318 16.5725 37.9318 10.5227 34.2046 6.79546C30.4775 3.06818 24.428 3.06818 20.7008 6.79546L19.5 7.99635L18.2992 6.79546C14.572 3.06818 8.5225 3.06818 4.79536 6.79546C1.06821 10.5227 1.06821 16.5725 4.79536 20.2998L19.5 35.005Z"
            fill="#B4171E"
          />
        </svg>
      </div>
    </div>
    <div
      class="css-ocrfjm"
    >
      <p
        class="chakra-text css-bq8qtg"
      >
        1
      </p>
      <div
        class="css-14d7ov9"
      >
        <svg
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 35.005L34.2046 20.2998C37.9318 16.5725 37.9318 10.5227 34.2046 6.79546C30.4775 3.06818 24.428 3.06818 20.7008 6.79546L19.5 7.99635L18.2992 6.79546C14.572 3.06818 8.5225 3.06818 4.79536 6.79546C1.06821 10.5227 1.06821 16.5725 4.79536 20.2998L19.5 35.005Z"
            fill="#B4171E"
          />
        </svg>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`CardPanel スナップショット/link 1`] = `
<DocumentFragment>
  <a
    class="css-3lw21s"
    href="/"
  >
    <div
      class="css-u8yv6t"
    >
      <p
        class="chakra-text css-bq8qtg"
      >
        1
      </p>
      <div
        class="css-14d7ov9"
      >
        <svg
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 35.005L34.2046 20.2998C37.9318 16.5725 37.9318 10.5227 34.2046 6.79546C30.4775 3.06818 24.428 3.06818 20.7008 6.79546L19.5 7.99635L18.2992 6.79546C14.572 3.06818 8.5225 3.06818 4.79536 6.79546C1.06821 10.5227 1.06821 16.5725 4.79536 20.2998L19.5 35.005Z"
            fill="#B4171E"
          />
        </svg>
      </div>
    </div>
    <div
      class="css-ocrfjm"
    >
      <p
        class="chakra-text css-bq8qtg"
      >
        1
      </p>
      <div
        class="css-14d7ov9"
      >
        <svg
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 35.005L34.2046 20.2998C37.9318 16.5725 37.9318 10.5227 34.2046 6.79546C30.4775 3.06818 24.428 3.06818 20.7008 6.79546L19.5 7.99635L18.2992 6.79546C14.572 3.06818 8.5225 3.06818 4.79536 6.79546C1.06821 10.5227 1.06821 16.5725 4.79536 20.2998L19.5 35.005Z"
            fill="#B4171E"
          />
        </svg>
      </div>
    </div>
  </a>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`CardPanel スナップショット/md 1`] = `
<DocumentFragment>
  <div
    class="css-1qxjl2e"
  >
    <div
      class="css-gh5eqk"
    >
      <p
        class="chakra-text css-11xd71r"
      >
        1
      </p>
      <div
        class="css-1ia5mzk"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 40 40"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 35.005L34.2046 20.2998C37.9318 16.5725 37.9318 10.5227 34.2046 6.79546C30.4775 3.06818 24.428 3.06818 20.7008 6.79546L19.5 7.99635L18.2992 6.79546C14.572 3.06818 8.5225 3.06818 4.79536 6.79546C1.06821 10.5227 1.06821 16.5725 4.79536 20.2998L19.5 35.005Z"
            fill="#B4171E"
          />
        </svg>
      </div>
    </div>
    <div
      class="css-pgpbif"
    >
      <p
        class="chakra-text css-11xd71r"
      >
        1
      </p>
      <div
        class="css-1ia5mzk"
      >
        <svg
          fill="none"
          height="24"
          viewBox="0 0 40 40"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 35.005L34.2046 20.2998C37.9318 16.5725 37.9318 10.5227 34.2046 6.79546C30.4775 3.06818 24.428 3.06818 20.7008 6.79546L19.5 7.99635L18.2992 6.79546C14.572 3.06818 8.5225 3.06818 4.79536 6.79546C1.06821 10.5227 1.06821 16.5725 4.79536 20.2998L19.5 35.005Z"
            fill="#B4171E"
          />
        </svg>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`CardPanel スナップショット/md 2`] = `
<DocumentFragment>
  <div
    class="css-vxlbt5"
  >
    <div
      class="css-u8yv6t"
    >
      <p
        class="chakra-text css-bq8qtg"
      >
        1
      </p>
      <div
        class="css-14d7ov9"
      >
        <svg
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 35.005L34.2046 20.2998C37.9318 16.5725 37.9318 10.5227 34.2046 6.79546C30.4775 3.06818 24.428 3.06818 20.7008 6.79546L19.5 7.99635L18.2992 6.79546C14.572 3.06818 8.5225 3.06818 4.79536 6.79546C1.06821 10.5227 1.06821 16.5725 4.79536 20.2998L19.5 35.005Z"
            fill="#B4171E"
          />
        </svg>
      </div>
    </div>
    <div
      class="css-ocrfjm"
    >
      <p
        class="chakra-text css-bq8qtg"
      >
        1
      </p>
      <div
        class="css-14d7ov9"
      >
        <svg
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.5 35.005L34.2046 20.2998C37.9318 16.5725 37.9318 10.5227 34.2046 6.79546C30.4775 3.06818 24.428 3.06818 20.7008 6.79546L19.5 7.99635L18.2992 6.79546C14.572 3.06818 8.5225 3.06818 4.79536 6.79546C1.06821 10.5227 1.06821 16.5725 4.79536 20.2998L19.5 35.005Z"
            fill="#B4171E"
          />
        </svg>
      </div>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
